pkg/solana/cache_test.go

```diff
@@ -106,8 +106,9 @@ func TestGetState(t *testing.T) {
 	}))
 	defer mockServer.Close()
 
+	reader := func() (client.AccountReader, error) { return testSetupReader(t, mockServer.URL), nil }
```

Comment:
why is this recreating a client/reader each time?

it would be cleaner to create a reader and then return it with a `getReader` function